### ISSUE TITLE
Support From & To annotations

### ISSUE THREAD

**Issue by sherifkayad** [pomerium/ingress-controller#95]
**Is your feature request related to a problem? Please describe.**

Running the Controller in the Operator mode and using an Nginx Ingress that DOESN'T terminate SSL (e.g. using an AWS ALB that handles that), requires not using an SSL block in the Ingress definition and hence the need to manually state the `From` piece of the policy.

Also we might need to get rid of the Nginx `auth-url` and `auth-signin` in favor of routing the traffic to the Pomerium proxy service (happily I can explain the main reason separately in a comment) and for that one would need the `To` piece.

**Describe the solution you'd like**
I would like to see the Ingress Controller supporting the following annotations:
- `ingress.pomerium.io/from`
- `ingress.pomerium.io/to`

**Describe alternatives you've considered**
Keep running the Operator and even forking and changing it beyond the deprecation notice from the project .. Currently we are running EKS v1.21 and we know that the Operator would stop working starting v1.22, however, if it needs be, we will fork it and change it because so far it has been doing a great job to be honest and the migration to the Ingress Controller isn't / wasn't trivial as the docs stated.

**Explain any additional use-cases**

Nothing.

**Additional context**

Nothing.

**Comment by travisgroth** [comment]
> Running the Controller in the Operator mode and using an Nginx Ingress that DOESN'T terminate SSL (e.g. using an AWS ALB that handles that), requires not using an SSL block in the Ingress definition and hence the need to manually state the `From` piece of the policy.

Can you elaborate on this?  I'm afraid I'm not following the use case and why you'd need to manually specify a `from`.  The `from` is implicit in the `host` parameter of an ingress rule, and the `to` is implicit in the `backend` of the path definition.  Why would you want them to be different than what the ingress is actually specifying?  Maybe a concrete example would help clarify.

**Comment by sherifkayad** [comment]
@travisgroth well I can try to explain .. but thinking about it again it might not be that of a great value to pursue this issue further ..
- Regarding the `from` part of the older Operator behavior was that if you were using an Ingress without a TLS (if an external Load Balancer is handling the TLS termination) the rule always gave 401 and we had to explicitly state a `From` to `https://app.mydomain.com`. I think this is not the case anymore with the Controller, as I re-tested that a couple of times. So the From piece can be discarded.
- Regarding the `to` part => as a workaround for the infinite redirects and using a Forward Auth with Nginx (`auth-url` and `auth-signin`) we found that if the Ingress intended to be secured can directly go to the `pomerium-proxy` service, everything works like a charm and one would see a correct `403` so we used that as a good workaround for the infinite redirects issue. In that case one would need to explicitly define `to` in order to map to the original service intended for the proxy.

To sum it up, having From & To (specially to) might give some flexibility .. however, if that's going to be an overkill we could discard that for now.

**Comment by travisgroth** [comment]
I'm not totally following the second use case but that sounds like a workaround at best.  Going to close this.  `to` and `from` should really come from the ingres spec.

**Comment by jawabuu** [comment]
Hey @travisgroth @wasaga kind request to re-introduce `ingress.pomerium.io/to`  annotation
As an optionaI annotation I don't believe it poses any conflict with the pomerium ingress controller.
The upside is that it gives greater flexibility in defining policies in other ingresses. Note that this was functional in pomerium operator.
I'll give you a practical example,
We are able to use forward auth with other ingresses see the traefik example for reference.

However, proxy mode is much more desired, and it can be achieved by defining policies as @sherifkayad noted.
Operator provided a way to define this policy on the ingress itself as below
```
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/ssl-proxy-headers: 'X-Forwarded-Proto: https'
    kubernetes.io/ingress.class: traefik-cert-manager
    kubernetes.io/tls-acme: "true"
    traefik.ingress.kubernetes.io/router.entrypoints: web, websecure
    traefik.ingress.kubernetes.io/router.tls: "true"
    ingress.pomerium.io/allowed_domains: '["domain.com"]'
    ingress.pomerium.io/to: http://whoami.whoami:80
    ingress.pomerium.io/tls_skip_verify: "true"
    ingress.pomerium.io/preserve_host_header: "true"
    ingress.pomerium.io/pass_identity_headers: "true"
  name: whoami
  namespace: whoami
spec:
  rules:
  - host: who.domain.com
    http:
      paths:
      - backend:
          service:
            name: pomerium-proxy
            port:
              number: 80
        pathType: ImplementationSpecific
  tls:
  - hosts:
    - who.domain.com
    secretName: whoami-tls
```
This was extremely convenient for situations where ingresses are tightly controlled as a matter of policy & in other cases where users simply want to use ingress controllers they are familiar with.
